rewrite-groovy/src/main/java/org/openrewrite/groovy/tree/G.java

```diff
@@ -1020,4 +1020,109 @@ public G.Unary withOperator(JLeftPadded<G.Unary.Type> operator) {
             }
         }
     }
+
+    @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+    @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
+    @RequiredArgsConstructor
+    @AllArgsConstructor(access = AccessLevel.PRIVATE)
+    final class DestructuringDeclaration implements G, Statement {
```

Comment:
I am wondering if it would make sense that the DestructuringDeclaration class would extend the VariableDeclarations class. For the Kotlin variant this is also _**not**_ done, but maybe it would make sense to do so? What do you think @shanman190 and @greg-at-moderne?